drkafka/src/main/java/com/pinterest/doctorkafka/api/MaintenanceApi.java

```diff
@@ -0,0 +1,53 @@
+package com.pinterest.doctorkafka.api;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.pinterest.doctorkafka.DoctorKafka;
+import com.pinterest.doctorkafka.KafkaClusterManager;
+
+@Path("/cluster/{clusterName}/admin/maintenance")
+@Produces({ MediaType.APPLICATION_JSON })
+@Consumes({ MediaType.APPLICATION_JSON })
+public class MaintenanceApi {
+
+  private DoctorKafka drKafka;
+
+  public MaintenanceApi(DoctorKafka drKafka) {
+    this.drKafka = drKafka;
+  }
+  
+  @GET
+  public boolean checkMaintenance(@PathParam("clusterName") String clusterName) {
```

Comment:
can we add `@Context HttpServletRequest req` parameter, and log the remote host, and user for sending the request? 

same for the other API calls.

Reply:
@yuyang08 request logging comes out of the box in Dropwizard, can we just leverage those ?

Reply:
@ambud  could you share some sample of Dropwizard logs?  Do we need extra setting in Dropwizard for that?

Reply:
Added explicit logging for maintenance mode. Dropwizard access log can be used for all other APIs.